File: 2013m./uzd1./sprendimai/uzd1.cpp
#include <fstream>
#include <vector>
#include <string>

using namespace std;

class Company {
public:
    string name;
    int x;
    int y;

    Company(string name, int x, int y) : name(name), x(x), y(y) {}
};

int getTravelDistance(const Company& company) {
    return (abs(company.x) + abs(company.y)) * 2;
}

int readData(vector<Company>& companies) {
    ifstream data("U1.txt");

    string dayInfo;
    int companyCount, maxKm;
    data >> companyCount >> maxKm;

    for (int i = 0; i < companyCount; ++i) {
        string companyName;
        int x, y;
        data >> companyName >> x >> y;
        companies.push_back(Company(companyName, x, y));
    }

    data.close();

    return maxKm;
}

void writeOutput(int companiesVisited, int km, string lastCompanyVisited) {
    ofstream output("U1rez.txt");
    output << companiesVisited << " " << km << " " << lastCompanyVisited;
    output.close();
}

int main() {
    vector<Company> companies;

    int maxKm = readData(companies);

    int i = 0;
    int km = 0;
    while (i < companies.size()) {
        int newKm = km + getTravelDistance(companies[i]);
        if (newKm > maxKm) {
            break;
        }
        km = newKm;
        i++;
    }

    writeOutput(i, km, companies[i - 1].name);

    return 0;
}

File: 2013m./uzd1./sprendimai/uzd1.py
class Company:
    def __init__(self, name, x, y):
        self.name = name
        self.x = x
        self.y = y

def getTravelDistance(company):
    return (abs(company.x) + abs(company.y)) * 2

def readData():
    data = open("U1.txt", "r")

    dayInfo = next(data).split(" ")
    companyCount = int(dayInfo[0])
    maxKm = int(dayInfo[1])

    companies = [None]*companyCount
    for i in range(companyCount):
        companyInfo = next(data)
        companyName = companyInfo[:10]
        companyInfo = companyInfo[11:].split(" ")
        companyInfo = [int(o) for o in companyInfo]

        companies[i] = Company(companyName, companyInfo[0], companyInfo[1])
    
    data.close()
    
    return maxKm, companies

def writeOutput(companiesVisited, km, lastCompanyVisited):
    output = open("U1rez.txt", "w")
    output.write(f"{companiesVisited} {km} {lastCompanyVisited}")
    output.close()

def main():
    maxKm, companies = readData()

    i = 0
    companyCount = len(companies)
    km = 0
    while(i < companyCount and km < maxKm):
        newKm = km + getTravelDistance(companies[i])
        if(newKm > maxKm):
            break
        km = newKm
        i += 1

    writeOutput(i, km, companies[i - 1].name)

if __name__ == "__main__":
    main()